### PR TITLE
Rework AutoFollowIT.testAutoFollowSearchableSnapshotsFails

### PR DESCRIPTION
**Pull request by tlrx**
The test `AutoFollowIT.testAutoFollowSearchableSnapshotsFails` was added in #70580 in order to test that mounted indices of a leader cluster arenot auto-followed in a follower cluster using CCR.

This test sometimes fails because it expects 2 indices to be followed (the `-regular` and the `-index` indices) but not the mounted one. This looks wrong as the `-index` index is deleted soon after it is snapshotted, and this index only exist to create a snapshot that can be later mounted as an index in the leader cluster.

This pull request changes the test so that the `-index` index, the repository and the snapshot are created at the beginning of the test. Then the test creates the mounted index and the regular one and can now asserts that only the regular one was auto-followed. That sounds more correct to me.

### CODE REVIEW

File: x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java [hunk 1]
```diff
@@ -805,33 +787,56 @@ public void testAutoFollowSearchableSnapshotsFails() throws Exception {
...
+                assertThat(eval("auto_follow_stats.recent_auto_follow_errors", response),
+                    hasSize(greaterThanOrEqualTo(1)));
+                assertThat(eval("auto_follow_stats.recent_auto_follow_errors.0.auto_follow_exception.reason", response),
+                    containsString("index to follow [" + mountedIndex + "] is a searchable snapshot index and cannot be used " +
+                        "for cross-cluster replication purpose"));
+                ensureYellow(regularIndex);
+                verifyDocuments(client(), regularIndex, 10);
+            });
+        } finally {
+            cleanUpLeader(List.of(regularIndex, mountedIndex), List.of(), List.of(autoFollowPattern));
```
Comment (fcofdez) [inline x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java:837]: I think we should remove the autoFollowPattern in the follower?

**Review by fcofdez — APPROVED** [review]
LGTM 👍 , I left a comment about cleanup

File: x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java (hunk 1 above)
Comment (tlrx) [reply x-pack/plugin/ccr/qa/multi-cluster/src/test/java/org/elasticsearch/xpack/ccr/AutoFollowIT.java:837]: Oh right, thanks! It should be removed on the follower, not the leader. I'll change that

**Comment by tlrx** [comment]
@elasticmachine run elasticsearch-ci/bwc

**Comment by tlrx** [comment]
Thanks Francisco

**Pull request merged**